PR for release

erocheleau opened the pull request:
Main features:
- The deploy package command now handles correctly the partial successes/failures.
- The deploy errors will be displayed using vscode Diagnostic feature.

Minor stuff:
- Added examples of tests to run on build.
- Configured Travis to run the tests on a new PR both in develop and master.

erocheleau committed:
Fixed the singleton config
+ Removed obligatory param security Token

erocheleau committed:
Deploy now supports partial success and
better error handling

erocheleau committed:
Deploy now supports partial success and
better error handling

erocheleau committed:
Removed TODO comment
Removed unused methods

erocheleau committed:
Removed TODO comment
Removed unused methods

erocheleau committed:
Removed empty line
Travis will build this?

erocheleau committed:
Removed empty line
Travis will build this?

erocheleau committed:
Travis where art thou?

erocheleau committed:
Travis where art thou?

erocheleau committed:
Merge pull request #86 from coveo/deploy-partial-successes

Deploy now supports partial success and better error handling and display! 🎉

erocheleau committed:
Added test for utils as an example (#89)

* Added test for utils as an example

* test travis

* change package json to include test command

* Modified utils.test.ts output

erocheleau committed:
Added test for utils as an example (#89)

* Added test for utils as an example

* test travis

* change package json to include test command

* Modified utils.test.ts output

erocheleau committed:
Fix words package deploy (#91)

* Fixed wording on package deploy

* Added a sort on the choosing of package.xml

* Actually sorty the package this time...

* Case sensitive?

* Changed interface + compile option in tsconfig

* Travis why?

* Windows file casing...

* Windows casing take 2, use git mv

erocheleau committed:
Fix words package deploy (#91)

* Fixed wording on package deploy

* Added a sort on the choosing of package.xml

* Actually sorty the package this time...

* Case sensitive?

* Changed interface + compile option in tsconfig

* Travis why?

* Windows file casing...

* Windows casing take 2, use git mv

erocheleau committed:
Merge branch 'develop' of https://github.com/coveo/vsforce into develop

erocheleau committed:
merge master into develop

lbenie reviewed (approved):
so much rebase

tedre191 commented on .travis.yml:
```diff
@@ -6,10 +6,22 @@ branches:
+  - linux
+
+before_install:
+  - if [ $TRAVIS_OS_NAME == "linux" ]; then
```
lol ?

lbenie replied:
on spawn 2 machines maintenant linux/osx
ca vient d'un exemple de microsoft IIRC

tedre191 commented on .travis.yml:
```diff
@@ -6,10 +6,22 @@ branches:
 install:
   - npm install typescript@1.8 -g
   - npm install
+  - npm run vscode:prepublish
```
ça va dans script

erocheleau replied:
non je l'ai change selon l'exemple de vscode non?

erocheleau replied:
Il est effectivement dans scripts...

erocheleau commented on .travis.yml:
```diff
@@ -6,10 +6,22 @@ branches:
     - develop
+
+os:
+  - osx
```
We should really remove the osx, since it puts our build in a queue to wait for an available osx machine and it takes about 30 minutes :(

lbenie replied:
yep let's do it

erocheleau committed:
Fix a bug when there is no security token in the config

erocheleau committed:
Merge pull request #94 from coveo/fix-no-securitytoken

Fix a bug when there is no security token in the config

lbenie reviewed (changes requested):
by default it should be set to ''

check package.json

🙉

erocheleau committed:
Fix for release 0.2.0
- No longer building on osx
- Default empty string value on string config params
- Reverted config.ts and added default value instead

Commit:
Merge pull request #95 from coveo/fix-before-0.2.0

Fix for release 0.2.0

tedre191 commented on package.json:
```diff
@@ -133,17 +127,17 @@
                 "vsforce.organization.username": {
                     "type": "string",
-                    "default": null,
+                    "default": "",
```
Nope. validation will fail if we passe "" as default for required configs

erocheleau replied:
Can't we check on length > 0 instead?

erocheleau replied:
null is definitely not an expected value for a string...

tedre191 replied:
we could, either way works

tedre191 commented on package.json:
```diff
@@ -133,17 +127,17 @@
                 "vsforce.organization.password": {
                     "type": "string",
-                    "default": null,
+                    "default": "",
```
same

erocheleau replied:
Can't we check on length > 0 instead?

erocheleau committed:
Fix-validation of null security token

tedre191 reviewed (approved):
:shipit:
